vnext/ReactUWP/Base/CoreNativeModules.cpp

```diff
@@ -130,12 +131,22 @@ std::vector<facebook::react::NativeModuleDescription> GetCoreModules(
 
   // AsyncStorageModule doesn't work without package identity (it indirectly depends on
   // Windows.Storage.StorageFile), so check for package identity before adding it.
-  if (HasPackageIdentity()) {
-    modules.emplace_back(
-        "AsyncLocalStorage",
-        []() { return std::make_unique<facebook::react::AsyncStorageModule>(L"asyncStorage"); },
-        MakeSerialQueueThread());
-  }
+  modules.emplace_back(
+      "AsyncLocalStorage",
+      [wpUwpInstance = std::weak_ptr(uwpInstance)]() -> std::unique_ptr<facebook::xplat::module::CxxModule> {
+        auto result = std::unique_ptr<facebook::xplat::module::CxxModule>();
+        if (HasPackageIdentity()) {
+          result = std::make_unique<facebook::react::AsyncStorageModule>(L"asyncStorage");
+        } else {
```

Comment:
>else { [](start = 10, length = 6)

Does it mean that UWP applications will go this conditional branch too?

Reply:
I'm sorry, I don't quite follow what you're asking... UWP apps have package identity so they'll continue to use `facebook::react::AsyncStorageModule>`.

Reply:
This code used to be UWP-only. I am not sure why the check was there. Would it be possible that UWP app returns false from the HasPackageIdentity() ?

Reply:
We added this a month or two ago; ReactUWP (really Microsoft.ReactNative, the old name makes it confusing :)) is intended to be usable by unpackaged apps.